taskcluster/script.py

```diff
@@ -203,22 +261,32 @@ def main():
     rc = None
     bytes_read = 0
     bytes_written = 0
-    proc = subprocess.Popen(extra_args,
-                            bufsize=0,
-                            env=env,
-                            stdout=subprocess.PIPE,
-                            stderr=subprocess.STDOUT)
-    while True:
-        line = proc.stdout.readline()
-        decoded_line = line.decode()
-        line_len = len(decoded_line)
-        bytes_read += line_len
-        rc = proc.poll()
-        if line:
-            bytes_written += sys.stdout.write(decoded_line)
-        if line_len == 0 and bytes_written == bytes_read and rc is not None:
-            break
-    print("script.py: command finished (bytes read: %s, bytes written: %s)" % (bytes_read, bytes_written))
+    dpi = DebugPrinter(device)
+    # timeout in x minutes
+    with timeout(TIMEOUT_MINUTES * 60, dpi):
```

Comment:
interesting use...

Reply:
> Let's make the <= changes. I definitely don't want this merged to master though. Let's just deploy this branch to a testing image to get the debugging information and we can move on from there. I'd like to schedule a meeting this coming week to talk about this script.py. I think we should consider making a number of changes but I'd like to work more closely with you on them.

Yeah, the previous change wasn't landed to master either.

Meeting sounds good.